### ISSUE TITLE
Incorrect interpretation of the field description

### ISSUE THREAD

**Issue by sergeyklay** [daveshanley/vacuum#239]
Hi,

vacuum says that `description` at [line **165**](https://github.com/sergeyklay/specmatic-testing-example/blob/073613ff9b41e70d3e05b6cd10f90983c4ca5dc3/contracts/documentation.yaml#L164-L166) is a duplicate of [line **62**](https://github.com/sergeyklay/specmatic-testing-example/blob/073613ff9b41e70d3e05b6cd10f90983c4ca5dc3/contracts/documentation.yaml#L61-L63)

Which is obviously a somewhat useless warning, due to the fact that these are fields of different models. Here, `description` field is not part of the OpenAPI syntax, this field describes an object attribute.

Log:
```
% vacuum lint -d contracts/documentation.yaml

██    ██  █████   ██████ ██    ██ ██    ██ ███    ███
██    ██ ██   ██ ██      ██    ██ ██    ██ ████  ████
██    ██ ███████ ██      ██    ██ ██    ██ ██ ████ ██
 ██  ██  ██   ██ ██      ██    ██ ██    ██ ██  ██  ██
  ████   ██   ██  ██████  ██████   ██████  ██      ██

version: latest | compiled: 2023-02-05 10:49:14 CET
🔗 https://quobix.com/vacuum | https://github.com/daveshanley/vacuum


 INFO  Linting against 42 rules: https://quobix.com/vacuum/rulesets/recommended


# Descriptions Issues

Line / Column | Severity | Message                                               | Path
(165:21)      | info     | Description at line `165` is a duplicate of line `62` | $.paths./v1/products/{id}.get.responses.400.content.applicat...
(182:21)      | info     | Description at line `182` is a duplicate of line `62` | $.paths./v1/products/{id}.get.responses.404.content.applicat...

# Examples Issues

Line / Column | Severity | Message                                                                     | Path
(133:15)      | warning  | Schema for `application/json` does not contain any examples or example data | $.paths./v1/products.get.responses.200
(157:15)      | warning  | Schema for `application/json` does not contain any examples or example data | $.paths./v1/products/{id}.get.responses.400
(174:15)      | warning  | Schema for `application/json` does not contain any examples or example data | $.paths./v1/products/{id}.get.responses.404

Category     | Errors | Warnings | Info
Descriptions | 0      | 0        | 2
Examples     | 0      | 3        | 0



          Linting passed, but with 3 warnings and 2 informs
```

Full OpeanAPI example:
```yaml
---
openapi: 3.0.3

info:
  version: '1.0'
  title: Products API
  description: Example API to test Specmatic features.

servers:
  - description: Localhost
    url: http://127.0.0.1:5000

tags:
  - name: products
    description: Products related API.

components:
  parameters:
    AcceptHeader:
      in: header
      name: Accept
      description: Accept Header
      schema:
        type: string
        enum: [application/json, '*/*']
      example: application/json
    IdParam:
      in: path
      name: id
      description: Entity ID param
      schema:
        type: integer
        format: int32
        minimum: 1
      required: true
      example: 42

  schemas:
    Product:
      title: Product
      allOf:
        - $ref: '#/components/schemas/ProductId'
        - $ref: '#/components/schemas/ProductDetails'
    ProductId:
      title: Product ID
      type: object
      properties:
        id:
          type: integer
          format: int32
          minimum: 1
          example: 42
      required: [id]
    ProductDetails:
      title: Product Details
      type: object
      properties:
        title:
          type: string
          example: iPhone 9
        description:
          type: string
          example: An apple mobile which is nothing like apple
        price:
          type: number
          minimum: 0
          example: 549
        discount:
          type: number
          minimum: 0.0
          example: 12.96
        rating:
          type: number
          minimum: 0.0
          maximum: 5.0
          example: 4.69
        stock:
          type: integer
          format: int32
          minimum: 0
          example: 856
        brand:
          type: string
          example: Apple, Inc.
        category:
          $ref: '#/components/schemas/ProductCategory'
      required:
        - title
        - description
        - price
        - discount
        - rating
        - stock
        - brand
        - category
    ProductCategory:
      title: Product Category
      type: string
      enum:
        - smartphones
        - laptops
      example: smartphones

paths:
  /v1/products:
    get:
      summary: List of all products.
      operationId: products.index
      tags: [products]
      parameters:
        - $ref: '#/components/parameters/AcceptHeader'
        - name: category
          description: Filter by product category.
          in: query
          required: false
          schema:
            type: string
          allowEmptyValue: true
          example: laptops
        - name: q
          description: Search term.
          in: query
          required: false
          schema:
            type: string
          allowEmptyValue: true
          example: samsung
      responses:
        '200':
          description: List of products
          content:
            application/json:
              schema:
                type: array
                items:
                  $ref: '#/components/schemas/Product'

  '/v1/products/{id}':
    get:
      summary: Get product.
      operationId: products.get
      tags: [products]
      parameters:
        - $ref: '#/components/parameters/AcceptHeader'
        - $ref: '#/components/parameters/IdParam'
      responses:
        '200':
          description: Get product
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Product'
        '400':
          description: Bad Request
          content:
            application/json:
              schema:
                type: object
                properties:
                  code:
                    type: integer
                    format: int32
                    example: 400
                  description:
                    type: string
                    example: Invalid product ID data type
                  name:
                    type: string
                    example: Bad Request
        '404':
          description: Not Found
          content:
            application/json:
              schema:
                type: object
                properties:
                  code:
                    type: integer
                    format: int32
                    example: 404
                  description:
                    type: string
                    example: Product not found
                  name:
                    type: string
                    example: Not Found
```

**Comment by daveshanley** [comment]
Interesting! I have a good idea why this is happening; it's a bug I think. Nice find

**Comment by daveshanley** [comment]
Resolved in `v0.0.52`

**Comment by sergeyklay** [comment]
Thank you!
